src/main/kotlin/pt/up/fe/ni/website/backend/repository/RoleRepository.kt

```diff
@@ -5,4 +5,6 @@ import org.springframework.stereotype.Repository
 import pt.up.fe.ni.website.backend.model.Role
 
 @Repository
-interface RoleRepository : CrudRepository<Role, Long>
+interface RoleRepository : CrudRepository<Role, Long> {
+    fun findByName(name: String): Role?
```

Comment:
Since name isn't unique, is there a way to specify that we only want the role associated with the most recent generation?

Reply:
I don't think it is possible without a custom query, but it only is used on role creation I think, but I'll change to make the according changes that @BrunoRosendo suggested and therefore I don't think findByName will be necessary at all.